sim/CMakeLists.txt

```diff
@@ -40,22 +47,77 @@ target_link_libraries(${PROJECT_NAME} PRIVATE glfw)
 set(GLAD_DIR "${LIB_DIR}/glad")
 add_library(glad "${GLAD_DIR}/src/glad.c")
 target_include_directories(glad PRIVATE "${GLAD_DIR}/include")
-target_include_directories(${PROJECT_NAME} PRIVATE "${GLAD_DIR}/include")
-target_link_libraries(${PROJECT_NAME} PRIVATE glad)
-target_include_directories("${PROJECT_NAME}_test" PRIVATE "${GLAD_DIR}/include")
-target_link_libraries("${PROJECT_NAME}_test" PRIVATE glad)
+foreach(target IN ITEMS ${TARGETS})
+    target_include_directories(${target} PRIVATE "${GLAD_DIR}/include")
+    target_link_libraries(${target} PRIVATE glad)
+    # On non-windows systems, GLAD requires libdl to be linked
+    if(NOT WIN32)
+        target_link_libraries(${target} PRIVATE dl)
+    endif(NOT WIN32)
+endforeach()
 
-# On non-windows systems, GLAD requires libdl to be linked
-if(NOT WIN32)
-target_link_libraries(${PROJECT_NAME} PRIVATE dl)
-target_link_libraries("${PROJECT_NAME}_test" PRIVATE dl)
-endif(NOT WIN32)
+# GRPC/Protobuf
+FetchContent_Declare(
+    grpc
+    GIT_REPOSITORY https://github.com/grpc/grpc.git
+    GIT_TAG        v1.33.2
+    GIT_PROGRESS   TRUE
+)
+# disable redudant dependencies targets
+# disable unused protobuf targets
+set(protobuf_BUILD_TESTS OFF)
+# disable unused grpc plugin targets
+set(gRPC_BUILD_CSHARP_EXT OFF)
+set(gRPC_BUILD_GRPC_CSHARP_PLUGIN OFF)
+set(gRPC_BUILD_GRPC_NODE_PLUGIN OFF)
+set(gRPC_BUILD_GRPC_OBJECTIVE_C_PLUGIN OFF)
+set(gRPC_BUILD_GRPC_PHP_PLUGIN OFF)
+set(gRPC_BUILD_GRPC_PYTHON_PLUGIN OFF)
+set(gRPC_BUILD_GRPC_RUBY_PLUGIN OFF)
+# disable abseil tests targets
+set(BUILD_TESTING OFF)
+FetchContent_MakeAvailable(grpc)
+# link protobuf/grpc libraries
+foreach(target IN ITEMS ${TARGETS})
+    target_include_directories(${target} PRIVATE "${PROTOBUF_DIR}/src")
+    target_link_libraries(${target} PRIVATE libprotobuf grpc++ grpc++_reflection)
+endforeach()
+
+# Compile and link protobuf c++ bindings from protobuf defitions
+get_filename_component(PROTOS_DIR "../protos" ABSOLUTE)
+# target directory to write protobuf bindings to
+set(PROTO_BINDS_DIR "${CMAKE_CURRENT_BINARY_DIR}/protos")
+file(MAKE_DIRECTORY ${PROTO_BINDS_DIR})
+# list protobuf defintions in protos/
+file(GLOB PROTO_DEFS "${PROTOS_DIR}/*.proto")
+# replace .proto with .pb.cc and .pb.h
+string(REPLACE "${PROTOS_DIR}" "${PROTO_BINDS_DIR}" PROTO_BINDS "${PROTO_DEFS}")
+string(REPLACE ".proto" ".pb.cc" PROTO_BIND_SOURCES "${PROTO_BINDS}")
```

Comment:
To keep extensions consistent, can we change this to `.cpp`?

Reply:
Unfortunately, I don't see a way other than telling CMake to rename the files from `.cc` to `.cpp`. I don't think that is worth the effort.